flow/packet-headers/arp.yaml

```diff
@@ -1,58 +1,78 @@
-openapi: 3.1.0
-
-info:
-  title: Arp packet header model
-  version: ^0.0.0
-
 components:
   schemas:
     Flow.Arp:
       description: ARP packet header
       type: object
       properties:
         hardware_type:
-          description: >-
-            Network link protocol type. 
-            Max length is 2 bytes.
-          $ref: './patterns.yaml#/components/schemas/Flow.Pattern'
+          x-pattern:
+            description: >-
+              Network link protocol type. 
+            format: hex
+            length: 16
+            default: '00'
+            features: [count, metric_group]
         protocol_type:
-          description: >-
-            The internetwork protocol for which the ARP request is intended. 
-            Max length is 2 bytes.
-          $ref: './patterns.yaml#/components/schemas/Flow.Pattern'
+          x-pattern:
+            description: >-
+              The internetwork protocol for which the ARP request is intended. 
+            format: hex
+            length: 16
+            default: '0800'
+            features: [count, metric_group]
         hardware_length:
-          description: >-
-            Length (in octets) of a hardware address. 
-            Max length is 1 bytes.
-          $ref: './patterns.yaml#/components/schemas/Flow.Pattern'
+          x-pattern:
+            description: >-
+              Length (in octets) of a hardware address. 
+            format: hex
```

Comment:
Would hw and protocol length be easily expressed using integer with. Max of 255, default: 6 and 4 for Ethernet and ipv4 respectively?

Reply:
For one byte values, this seems safe to use integer with max constraints since there is no endian issues

Reply:
format isn't a rigid spec - its used to enhance the type - in this case its critical that we define the length as bits or bytes (long string lengths expressed as bits isnt of much use)